tests/model/write_model_test.go

```diff
@@ -0,0 +1,298 @@
+package model
+
+import (
+	"context"
+	"testing"
+
+	parser "github.com/craigpastro/openfga-dsl-parser/v2"
+	"github.com/openfga/openfga/cmd"
+	"github.com/openfga/openfga/pkg/typesystem"
+	"github.com/openfga/openfga/tests"
+	"github.com/stretchr/testify/require"
+	pb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+	"google.golang.org/grpc/status"
+)
+
+var tt = map[string]struct {
+	model string
+	code  int
+}{
+	"case6": {
+		model: `
+		type user
+		type group
+		  relations
+			define group as group from group
+		`,
+		code: 2056,
+	},
+	"case9": {
+		model: `
+		type user
+		type org
+		  relations
+			define member: [user] as self
+		type group
+		  relations
+			define parent: [org] as self
+			define viewer as viewer from parent
+		`,
+		code: 2056,
+	},
+	"case10": {
+		model: `
+		type user
+		type group
+		  relations
+			define parent: [group] as self
+			define viewer as reader from parent
```

Comment:
Could we add a test where we have a model like

```
type user

type group
  relations
    define parent: [group, team] as self
    define viewer as reader from parent
    

type team
  relations
    define reader: [user] as self
```

and assert that it works just fine? So long as at least one types involved in the tupleset relation defines the computed_userset relationship, then it should be fine yeah?

Reply:
Nice test! This failed originally, so got a fix in.